### DISCUSSION TITLE
New instruction for inter-CTA barrier in future GPUs?

### DISCUSSION THREAD

**Discussion by wzhcz8902** [NVIDIA/cutlass#1502] [Ideas]
https://github.com/NVIDIA/cutlass/blob/5c447dd84f8ae0e1d48ff9a2eae26ce8c4958101/include/cutlass/semaphore.h#L52
I saw similar synchronization mechanism using semaphores between CTAs in cub, do you think it might be good to have an instruction that does this, or will this be a hardware feature in the future generation of GPUs?

**Comment by thakkarV** [comment]
Hopper adds a bunch of stuff for accelerated intern cluster sync
